azblob/list.go

```diff
@@ -16,12 +16,23 @@ func (azp *Storer) Count(ctx context.Context, tagsFilter string, opts ...Option)
 
 	logger.Sugar.Debugf("Count")
 
-	r, err := azp.FilteredList(ctx, tagsFilter, opts...)
-	if err != nil {
-		return 0, err
-	}
-
-	return int64(len(r.Items)), nil
+	var count int64
+	var m string
+	opts = append(opts, WithListMarker(&m))
+
+	for {
+		r, err := azp.FilteredList(ctx, tagsFilter, opts...)
+		if err != nil {
+			return 0, err
+		}
+		count += int64(len(r.Items))
+		logger.Sugar.Debugf("Count %d (%v)", count, r.Marker)
+		if r.Marker == nil {
```

Comment:
I think this may need to be `r.Marker == nil || *r.Marker == ""` have a look at the other implementations to see what they do

Reply:
I made the fix but it did not help. Serhiy has said the blobs caps limit tests must be rewritten...

Reply:
I am pretty sure you do need to check for marker is ""?  I found that problem elsewhere.

Reply:
old code